tool/tsh/tsh.go

```diff
@@ -3331,6 +3331,21 @@ var defaultWebProxyPorts = []int{
 	defaults.HTTPListenPort, teleport.StandardHTTPSPort,
 }
 
+// Generate error message from attempting hosts at different ports for web proxy
+func proxyHostsErrorMsgDefault(proxyAddress string, ports []int) (errorMessage string) {
+	var addresses = ""
+	for i := 0; i < len(ports); i++ {
+		addresses += fmt.Sprintf("%v:%v", proxyAddress, ports[i])
+		if (i + 1) < len(ports) {
+			addresses += " or "
+		}
+
+	}
+	msg := fmt.Sprintf("Teleport proxy not available at proxy address %s. "+
+		"Confirm address and connectivity.", addresses)
+	return msg
+}
```

Comment:
nit: Use godoc style, use a bytes.Buffer to "concatenate" strings, use for-range:

```suggestion
// proxyHostsErrorMsgDefault returns the error message from attempting hosts at
// different ports for the Web Proxy.
func proxyHostsErrorMsgDefault(proxyAddress string, ports []int) string {
	buf := &bytes.Buffer{}
	buf.WriteString("Teleport proxy not available at proxy address ")
	buf.WriteString(proxyAddress)
	buf.WriteString(". Confirm address and connectivity. ")

	for i, port := range ports {
		if i > 0 {
			buf.WriteString(" or ")
		}
		buf.WriteString(proxyAddress)
		buf.WriteString(":")
		buf.WriteString(strconv.Itoa(port))
	}

	return buf.String()
}
```

(Please give this a test, it looks alright to me but I just wrote it down in the comment.)